jme3-materialeditor/src/com/jme3/gde/materialdefinition/editor/ShaderNodePanel.java

```diff
@@ -194,15 +217,18 @@ protected void updateType() {
                 break;
             case Attribute:
                 header.setIcon(Icons.attrib);
-                setNameAndTitle("Attribute"); // sets text _and_ tooltip the same
+                name = "Attribute";
+                setTitle(outputLabels.get(0).getText() ); // sets text _and_ tooltip the same
                 break;
             case WorldParam:
                 header.setIcon(Icons.world);
-                setNameAndTitle("WorldParam");
+                name = "WorldParam";
+                setTitle(outputLabels.get(0).getText());
                 break;
             case MatParam:
                 header.setIcon(Icons.mat);
-                setNameAndTitle("MatParam");
+                name = "MatParam";
+                setTitle(outputLabels.get(0).getText());
                 break;
```

Comment:
Separating name and title for single output nodes. It allows for having the param name in the header.